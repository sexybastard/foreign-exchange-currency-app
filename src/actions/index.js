export const setAmount = payload => ({
  type: 'SET_AMOUNT',
  payload
})

export const setRates = payload => ({
  type: 'SET_RATES',
  payload
})

export const setRatesNameMap = payload => ({
  type: 'SET_RATES_NAME_MAP',
  payload
})

export const addCurrency = payload => ({
  type: 'ADD_CURRENCY',
  payload
})

export const deleteCurrency = payload => ({
  type: 'DELETE_CURRENCY',
  payload
})