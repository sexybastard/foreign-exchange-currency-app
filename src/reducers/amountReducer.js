const amount = (state = [], action) => {
  switch (action.type) {
    case 'SET_AMOUNT':
      return action.payload
    default:
      return state
  }
}

export default amount