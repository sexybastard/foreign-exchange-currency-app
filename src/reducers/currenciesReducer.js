const currencies = (state = [], action) => {
  switch (action.type) {
    case 'ADD_CURRENCY':
      return [...state, action.payload]
    case 'DELETE_CURRENCY':
      return [...state.slice(0, state.indexOf(action.payload)), ...state.slice(state.indexOf(action.payload) + 1)]
    default:
      return state
  }
}

export default currencies