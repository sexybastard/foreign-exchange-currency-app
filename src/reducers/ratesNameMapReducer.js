const ratesNameMap = (state = [], action) => {
  switch (action.type) {
    case 'SET_RATES_NAME_MAP':
      return action.payload
    default:
      return state
  }
}

export default ratesNameMap