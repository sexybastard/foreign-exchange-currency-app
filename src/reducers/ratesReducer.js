const rates = (state = {}, action) => {
  switch (action.type) {
    case 'SET_RATES':
      return action.payload
    default:
      return state
  }
}

export default rates