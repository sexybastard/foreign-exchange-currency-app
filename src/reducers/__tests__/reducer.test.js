import amountReducer from '../amountReducer'
import currenciesReducer from '../currenciesReducer'
import ratesNameMapReducer from '../ratesNameMapReducer'
import ratesReducer from '../ratesReducer'
import { setAmount, setRates, setRatesNameMap, addCurrency, deleteCurrency } from '../../actions';

const ratesNameMap = {
  PHP: 'Philippine Piso',
  HUF: 'Hungarian Forint',
  IDR: 'Indonesian Rupiah',
  TRY: 'Turkish Lira',
  RON: 'Romanian Leu',
  ISK: 'Icelanding Krona',
  ILS: 'Israeli Shekel',
  CNY: 'Chinese Yuan',
  USD: 'United States Dollar',
  EUR: 'Euro',
  PLN: 'Polish Zloty',
  GBP: 'Pound Sterling',
  CAD: 'Canadian Dollar',
  AUD: 'Australian Dollar',
  MYR: 'Malaysian Ringgit',
  NZD: 'New Zealand Dollar',
  CHF: 'Swiss Franc',
  HRK: 'Croatian Kuna',
  SGD: 'Singapore Dollar',
  DKK: 'Danish Krone',
  BGN: 'Bulgarian Lev',
  CZK: 'Czech Koruna',
  BRL: 'Brazilian Rel',
  JPY: 'Japanese Yen',
  KRW: 'South Korean Won',
  INR: 'Indian Rupee',
  SEK: 'Swedish Krona',
  MXN: 'Mexican Peso',
  RUB: 'Russian Ruble',
  HKD: 'Hong Kong Dollar',
  ZAR: 'South African Rand',
  THB: 'Thai Baht',
  NOK: 'Norwegian Krone'
};

describe("reducers: ", () => {
  it('test amountReducer', () => {
    expect(amountReducer(undefined, setAmount(100))).toMatchSnapshot();
  })
  it('test currenciesReducer', () => {
    expect(currenciesReducer(undefined, addCurrency('IDR'))).toMatchSnapshot();
    expect(currenciesReducer(undefined, deleteCurrency('IDR'))).toMatchSnapshot();
  })
  it('test ratesNameMapReducer', () => {
    expect(ratesNameMapReducer(undefined, setRatesNameMap(ratesNameMap))).toMatchSnapshot();
  })
  it('test ratesReducer', () => {
    expect(ratesReducer(undefined, setRates('IDR'))).toMatchSnapshot();
  })
});