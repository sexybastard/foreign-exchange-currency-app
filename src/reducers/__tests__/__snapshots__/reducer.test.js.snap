// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers:  test amountReducer 1`] = `100`;

exports[`reducers:  test currenciesReducer 1`] = `
Array [
  "IDR",
]
`;

exports[`reducers:  test currenciesReducer 2`] = `Array []`;

exports[`reducers:  test ratesNameMapReducer 1`] = `
Object {
  "AUD": "Australian Dollar",
  "BGN": "Bulgarian Lev",
  "BRL": "Brazilian Rel",
  "CAD": "Canadian Dollar",
  "CHF": "Swiss Franc",
  "CNY": "Chinese Yuan",
  "CZK": "Czech Koruna",
  "DKK": "Danish Krone",
  "EUR": "Euro",
  "GBP": "Pound Sterling",
  "HKD": "Hong Kong Dollar",
  "HRK": "Croatian Kuna",
  "HUF": "Hungarian Forint",
  "IDR": "Indonesian Rupiah",
  "ILS": "Israeli Shekel",
  "INR": "Indian Rupee",
  "ISK": "Icelanding Krona",
  "JPY": "Japanese Yen",
  "KRW": "South Korean Won",
  "MXN": "Mexican Peso",
  "MYR": "Malaysian Ringgit",
  "NOK": "Norwegian Krone",
  "NZD": "New Zealand Dollar",
  "PHP": "Philippine Piso",
  "PLN": "Polish Zloty",
  "RON": "Romanian Leu",
  "RUB": "Russian Ruble",
  "SEK": "Swedish Krona",
  "SGD": "Singapore Dollar",
  "THB": "Thai Baht",
  "TRY": "Turkish Lira",
  "USD": "United States Dollar",
  "ZAR": "South African Rand",
}
`;

exports[`reducers:  test ratesReducer 1`] = `"IDR"`;
